benchmark_runner/main/main.py

```diff
@@ -60,22 +59,26 @@
 
 
 @logger_time_stamp
-def azure_cluster_start_stop():
+def azure_cluster_operations():
     """
-    This method starts/ stops azure cluster
-    :return:
+    This method starts/stops Azure cluster according to input parameter azure_cluster_operation
+    @return:
     """
-    azure_operation = AzureOperations(azure_clientid=environment_variables_dict.get('azure_clientid', ''),
+    azure_operation = AzureOperations(
+                                      azure_clientid=environment_variables_dict.get('azure_clientid', ''),
                                       azure_secret=environment_variables_dict.get('azure_secret', ''),
                                       azure_tenantid=environment_variables_dict.get('azure_tenantid', ''),
                                       azure_subscriptionid=environment_variables_dict.get('azure_subscriptionid', ''),
-                                      azure_resource_group_name=environment_variables_dict.get(
-                                          'azure_resource_group_name', ''))
+                                      azure_resource_group_name=environment_variables_dict.get('azure_resource_group_name', ''),
+                                      kubeadmin_password=environment_variables_dict.get('kubeadmin_password', '')
+                                      )
     azure_vm_name = (environment_variables_dict.get('azure_vm_name', ''))
-    if azure_cluster_start:
-        logger.info(azure_operation.start_vm(vm_name=azure_vm_name))
-    elif azure_cluster_stop:
-        logger.info(azure_operation.stop_vm(vm_name=azure_vm_name))
+    if azure_cluster_operation == 'START':
+        logger.info('Start cluster with verification')
+        azure_operation.start_cluster_with_verification(vm_name=azure_vm_name)
+    elif azure_cluster_operation == 'STOP':
+        logger.info('Stop cluster')
+        azure_operation.stop_vm(vm_name=azure_vm_name)
```

Comment:
There also should be a branch for an invalid value.

Reply:
Fixed